src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -201,14 +204,9 @@ ApplicationManagerImpl::ApplicationManagerImpl(
                              {TYPE_SYSTEM, "System"},
                              {TYPE_ICONS, "Icons"}};
 
-  sync_primitives::AutoLock lock(timer_pool_lock_);
-  TimerSPtr clearing_timer(std::make_shared<timer::Timer>(
-      "ClearTimerPoolTimer",
-      new TimerTaskImpl<ApplicationManagerImpl>(
-          this, &ApplicationManagerImpl::ClearTimerPool)));
   const uint32_t timeout_ms = 10000u;
-  clearing_timer->Start(timeout_ms, timer::kSingleShot);
-  timer_pool_.push_back(clearing_timer);
+  clear_pool_timer_.Start(timeout_ms, timer::kSingleShot);
```

Comment:
It seems that `ApplicationManagerImpl::ClearTimerPool` will only ever be called one time, 10 seconds after the Application Manager is constructed.

Do you agree that `ApplicationManagerImpl::ClearTimerPool` should be called more often? I propose increasing the timeout and using `timer::kPeriodic`.

Reply:
@iCollin good point. Please take look. https://github.com/smartdevicelink/sdl_core/pull/3202/commits/4b7b3eae15682db636f68ef82e8afc2d1ffbbbc1
`kSingleShot` was introduced in https://github.com/smartdevicelink/sdl_core/commit/f6ebfe592402a87fe7118f0dbe65f8c46102e4e8
And this looks like a defect. We will run a regression testing with new changes.
How do you think, is this PR is a good place for this fix? Or it's better to create a new separate issue?

Reply:
I think this PR is a fine place for the fix, especially since we are moving one timer to a member variable.